### PR TITLE
Obsolete platform and always write csharp

### PR DESCRIPTION
**Pull request by SimonCropp**
related to #1590

### CODE REVIEW

File: src/Sentry/Exceptions/SentryStackFrame.cs [hunk 1]
```diff
@@ -160,7 +161,7 @@ public void WriteTo(Utf8JsonWriter writer, IDiagnosticLogger? logger)
             writer.WriteStringIfNotWhiteSpace("context_line", ContextLine);
             writer.WriteBooleanIfNotNull("in_app", InApp);
             writer.WriteStringIfNotWhiteSpace("package", Package);
-            writer.WriteStringIfNotWhiteSpace("platform", Platform);
+            writer.WriteString("platform", Constants.Platform);
```
Comment (SimonCropp) [inline src/Sentry/Exceptions/SentryStackFrame.cs:164]: @bruno-garcia @mattjohnsonpint if we always write `csharp` for the root event, can `"platform"` be omitted from stackframe and transaction?

File: src/Sentry/Exceptions/SentryStackFrame.cs [hunk 2]
```diff
@@ -104,7 +104,8 @@ public sealed class SentryStackFrame : IJsonSerializable
         /// <summary>
         /// This can override the platform for a single frame. Otherwise the platform of the event is assumed.
         /// </summary>
-        public string? Platform { get; set; }
```
Comment (mattjohnsonpint) [inline src/Sentry/Exceptions/SentryStackFrame.cs:107]: This one is different, as it's an override that's only needed when the platform is different than the one in the event.  We shouldn't set it to `Constants.Platform`.

I also don't think we will obsolete it completely, as we'll need it if we bundle the native SDK (as we will likely need to do for MAUI).  Though, I could see setting it `internal` in the future, which would effectively obsolete it.

File: test/Sentry.Tests/Protocol/Exceptions/SentryStackFrameTests.cs [hunk 3]
```diff
@@ -22,7 +22,6 @@ public void SerializeObject_AllPropertiesSetToNonDefault_SerializesValidObject()
             Vars = { { "var", "val" } },
             FramesOmitted = { 1, 2 },
             Package = "Package",
-            Platform = "Platform",
```
Comment (mattjohnsonpint) [inline test/Sentry.Tests/Protocol/Exceptions/SentryStackFrameTests.cs:25]: Lets keep this test as it was, to ensure we can override the platform for a single stack frame.

File: src/Sentry/Exceptions/SentryStackFrame.cs (hunk 1 above)
Comment (mattjohnsonpint) [reply src/Sentry/Exceptions/SentryStackFrame.cs:164]: It always needs to be sent to sentry, for all events including both errors and transactions.  But on stack frames, it should only be included if that frame originates from another platform.  So, if we end up bundling the native SDKs at some point, the stack frames captured by those SDKs would be different.  Thus, this line should be reverted to as it was, and we should keep `StackFrame.Platform` null by default.

**Review by mattjohnsonpint — CHANGES_REQUESTED** [review]
Please update stack frame as indicated.  Thanks.

File: test/Sentry.Tests/Internals/MonoSentryStackTraceFactoryTests.cs [hunk 4]
```diff
@@ -33,7 +33,6 @@ public void Create_UnityIl2Cpp_LeavesOutZeroedOutPath()
             Assert.Null(actual!.Frames[i].LineNumber);
             Assert.Null(actual!.Frames[i].InstructionOffset);
             Assert.Null(actual!.Frames[i].Package);
-            Assert.Null(actual!.Frames[i].Platform);
```
Comment (mattjohnsonpint) [inline test/Sentry.Tests/Internals/MonoSentryStackTraceFactoryTests.cs:36]: I think this one needs to be put back?

**Review by mattjohnsonpint — APPROVED** [review]

**Pull request merged**